cfme/tests/networks/nuage/test_events_are_triggered.py

```diff
@@ -61,6 +65,53 @@ def with_nuage_sandbox(networks_provider):
     logger.info('Destroyed sandbox enterprise {} ({})'.format(enterprise.name, enterprise.id))
 
 
+@pytest.fixture
+def targeted_refresh(merkyl_inspector):
+    """
+    This fixture tests whether targeted refresh was triggered for given targets.
+    It basically tails evm.log to see whether a line like this has occured:
+
+    ```
+    [Collection of targets with id: [{:ems_ref=>"b35d3afe-6f19-4da8-b1ee-b79de433cace"}, ... ]]
+    ```
+    for each target and raises an error if not.
+
+    Usage:
+
+        def test_something(targeted_refresh)
+                trigger_targeted_refresh() <- some function that will trigger
+                                              targeted refresh
+                with targeted_refresh.target_timeout():
+                    targeted_refresh.register_target('ref-123456', 'Subnet named TEST')
+                    targeted_refresh.register_target('ref-aaaabb', 'Router named TEST')
+    """
+    evm_log = '/var/www/miq/vmdb/log/evm.log'
+    needle_template = r'^.*Collection of targets with id.*:ems_ref=>"{}".*.$'
+    merkyl_inspector.add_log(evm_log)
+    targets = set()  # { (ems_ref, comment), (ems_ref, comment), ... }
+
+    def check_log():
```

Comment:
So what check_log currently does is it fetches evm.log content from the appliance for each of the 10 targets in our test and attempts to find needle in it. This means we transfer the evm.log content 10x every 5 seconds! Please consider refactoring this function to only fetch the evm.log once every 5 seconds, for example:

```python
def check_log():
        logger.info('Looking for %s needles in evm.log: %s', len(targets), [t[1] for t in targets])
        content = merkyl_inspector.get_log(evm_log)
        for target in set(targets):
            if target[0].search(content):
                logger.info('Found needle %s', target[1])
                targets.remove(target)
        return len(targets) == 0
```